ethexe/runtime/common/src/schedule.rs

```diff
@@ -67,15 +70,22 @@ impl<S: Storage> TaskHandler<Rfm, Sd, Sum> for Handler<'_, S> {
         0
     }
 
-    fn send_user_message(&mut self, stashed_message_id: MessageId, program_id: ProgramId) -> u64 {
+    fn send_user_message(
+        &mut self,
+        stashed_message_id: MessageId,
+        (program_id, dispatch_origin): (ProgramId, Origin),
```

Comment:
you don't need dispatch origin here since you have just taken dispatch from stash - so you have its origin

Reply:
btw I'd open an issue to keep in Mailbox no only value, but some additional info about mailboxed message: maybe whole dispatch / maybe just payload + value + origin so we won't need any changes in journal notes processing

Reply:
Fixed

Reply:
> btw I'd open an issue to keep in Mailbox no only value, but some additional info about mailboxed message: maybe whole dispatch / maybe just payload + value + origin so we won't need any changes in journal notes processing

Will add in #4460 PR